### PR TITLE
Migrating native sim from GCC to Clang.

### PR DESCRIPTION
**Pull request by kuzminrobin**
A set of 3 PRs: [Q#RT](https://github.com/microsoft/qsharp-runtime/pull/889), [QDK](https://github.com/microsoft/qdk/pull/194), [IQ#](https://github.com/microsoft/iqsharp/pull/553) repos (see also [quantum-docs PR](https://github.com/MicrosoftDocs/quantum-docs/pull/427)).
A set of the latter 2 are expected to require a branch-specific e2e build or a merge of the former one first.

### CODE REVIEW

**Comment by kuzminrobin** [comment]
Please review.

**Review by idavis — APPROVED** [review]

File: images/iqsharp-base/Dockerfile [hunk 1]
```diff
@@ -20,7 +20,7 @@ RUN apt-get update && \
                        # Dependencies for the Quantum Development Kit.
                        # Note that we install them here to minimize the number
                        # of layers.
-                       libgomp1 \
+                       libomp-7-dev \
```
Comment (cgranade) [inline images/iqsharp-base/Dockerfile:23]: Why is this needed in the iqsharp-base Dockerfile? That indicates to me that even with static linking, there's a new requirement passed onto users.

**Review by cgranade — CHANGES_REQUESTED** [review]
Marking as request changes to make sure that we don't add additional requirements to users without double-checking first.

File: images/iqsharp-base/Dockerfile (hunk 1 above)
Comment (kuzminrobin) [reply images/iqsharp-base/Dockerfile:23]: Static linking attempt has failed. I failed to find the `libomp.a` static library, it wasn't installed as a part of libomp package (I didn't try to build `libomp.a` from sources). 
Also, IIRC, there was some hypothetical risk with licensing, if we provide libomp together with our native simulator. But _installing_ the libomp had no any licensing issues.
> Why is this needed in the iqsharp-base Dockerfile?

Because the native simulator dynamic library is built on _ubuntu-latest_, where libomp is already installed, but the  iqsharp-base Docker image (that uses the native simulator dynamic library) is run on Debian Buster (Debian 10), where the libomp is not installed. So I added the installation of libomp on Debian Buster.

Comment (idavis) [reply images/iqsharp-base/Dockerfile:23]: @kuzminrobin can we try `libomp5-7`? That should just be the libomp runtime.

**Review by cgranade — APPROVED** [review]

**Comment by cgranade** [comment]
Approved, but we should be sure to document this change; users who work directly on Linux or in custom containers may need to update from their previous dependency.

**Comment by kuzminrobin** [comment]
@cgranade, any direct links to the documentation files that I need to update are more than welcome. If you mean just the QDK Release Notes (of January 2022 release) then feel free to reply with "relnotes".

**Comment by cgranade** [comment]
> @cgranade, any direct links to the documentation files that I need to update are more than welcome. If you mean just the QDK Release Notes (of January 2022 release) then feel free to reply with "relnotes".

I don't have any off the top of my head, sorry, but some of the install docs used to mention the libgomp1 dependency; searching for if there is still such a mention may be helpful, perhaps?

**Comment by kuzminrobin** [comment]
@cgranade, 
> some of the install docs used to mention the libgomp1 dependency; searching for if there is still such a mention may be helpful, perhaps?

See https://github.com/MicrosoftDocs/quantum-docs/pull/427.

File: conda-recipes/iqsharp/build.ps1 [hunk 2]
```diff
@@ -54,3 +54,14 @@ Push-Location $TargetDirectory
...
+
+Write-Host "## Installing OpenMP support into library path. ##"
+if ($IsLinux) {
+    Copy-Item -Verboese `
+        (Join-Path (Join-Path (Join-Path $TargetDirectory "runtimes") $RuntimeID) "libomp.so.*") `
+        (Join-Path (Join-Path $Env:PREFIX "lib") "x86_64-linux-gnu")
+} elseif ($IsMacOS) {
+    Copy-Item -Verboese `
+        (Join-Path (Join-Path (Join-Path $TargetDirectory "runtimes") $RuntimeID) "libomp.dylib") `
+        (Join-Path (Join-Path $Env:PREFIX "lib") "x86_64-darwin-macho")
```
Comment (kuzminrobin) [inline conda-recipes/iqsharp/build.ps1:66]:
```suggestion
        (Join-Path (Join-Path $Env:PREFIX "lib") "x86_64-darwin-machos")    
```
?
